microbench/src/main/java/io/netty/microbench/concurrent/BurstCostExecutorsBenchmark.java

```diff
@@ -0,0 +1,140 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.microbench.concurrent;
+
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import io.netty.util.concurrent.DefaultEventExecutor;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class BurstCostExecutorsBenchmark extends AbstractMicrobenchmark {
+
+    enum ExecutorType {
+        defaultEventExecutor,
+        juc,
+        nioEventLoop
+    }
+
+    @Param({ "1", "10" })
+    int burstLength;
+    @Param({ "defaultEventExecutor", "juc", "nioEventLoop" })
+    String executorType;
+
+    ExecutorService executor;
+    ExecutorService executorToShutdown;
+
+    @Setup
+    public synchronized void setup() {
+        ExecutorType type = ExecutorType.valueOf(executorType);
+        switch (type) {
+        case defaultEventExecutor:
+            executor = new DefaultEventExecutor();
+            executorToShutdown = executor;
+            break;
+        case juc:
+            executor = Executors.newSingleThreadScheduledExecutor();
+            executorToShutdown = executor;
+            break;
+        case nioEventLoop:
+            NioEventLoopGroup group = new NioEventLoopGroup(1);
+            group.setIoRatio(1);
+            executor = group.next();
+            executorToShutdown = group;
+            break;
+        }
+
+    }
+
+    @TearDown
+    public synchronized void tearDown() {
+        executorToShutdown.shutdown();
+    }
+
+    @State(Scope.Thread)
+    public static class PerThreadState {
+        private static final AtomicIntegerFieldUpdater<PerThreadState> DONE_UPDATER =
+                AtomicIntegerFieldUpdater.newUpdater(PerThreadState.class, "completed");
+        //thanks to JMH it would be padded avoiding false-sharing for free
+        private volatile int completed = 0;
+
+        private final Runnable completeTask = new Runnable() {
+            @Override
+            public void run() {
+                DONE_UPDATER.lazySet(PerThreadState.this, completed + 1);
```

Comment:
can you add a comment why lazySet is good enough ? Like explain we dont need to have it act as a full memory barrier etc... Also for the other case below.

Reply:
Added: I haven't noticed that it wasn't that easy to understand it...good point!